doc/services/ipc/ipc_service/ipc_service.rst

```diff
@@ -0,0 +1,251 @@
+.. _ipc_service:
+
+IPC service
+###########
+
+.. contents::
+    :local:
+    :depth: 2
+
+The IPC service API provides an interface to exchange the data between two
+domains/CPUs. Messages are exchanged between the domains through a shared
+memory region. Domains signal to each other data has been written with use of
+IPM devices.
+
+IPC service communication channel consist of the instance and the endpoint
+associated with an instance. An instance is the external representation of a
+physical communication channel between two domains/CPUs. The actual
+implementation and internal representation of the instance is peculiar to each
+backend.
+
+An individual instance is not used to send data between domains/CPUs. To send
+and receive the data, the user have to create (register) an endpoint in the
+instance. This allows to connect two domains of interest.
+
+It's possible to have zero or multiple endpoints in one single instance, each
+one is used to exchange data. Possibly with different priorities. Endpoint
+prioritization and multi instance ability highly depends on the backend used.
+
+The creation of the instances is left to the backend - usually at init time.
+Registration of the endpoints is left to the user - usually at run time.
+
+The API does not mandate a way for the backend to create instances but it is
+strongly recommended to use the DT to retrieve the configuration parameters for
+an instance. Currently each backend define its own DT compatible that is used
+to configure interface at boot time.
+
+The endpoint is an entity the user must use to send and receive data between
+two domains (connected by the instance). An endpoint is always associated to an
+instance.
+
+Usage
+*****
+
+Simple data exchange
+====================
+
+In order to send data between domains/CPUs an endpoint must be registered onto
+an instance. Note that it is up to the backend if before registering an
+endpoint instance must be opened with :c:func:`ipc_service_open_instance`
+function.
```

Comment:
uhm, this is a bit controversial. The user must always call `ipc_service_open_instance()`, it's not really dependent on the backend implementation.

Reply:
There might be some misunderstanding from my side. In fact in we have:
```
int ipc_service_open_instance(const struct device *instance)
{
	const struct ipc_service_backend *backend;
        [...]
	backend = (const struct ipc_service_backend *) instance->api;
        [...]
	if (!backend->open_instance) {
		/* maybe not needed on backend */
		return 0;
	}
```

In fact you can live without calling it. But i understand that we should always do it for the sake of consistency.

Reply:
yeah, the idea is that you want to always have that so you don't have to think which backend you are using and adapting your code.

Reply:
You are right. I will rephrase this sentence.